src/types.rs

```diff
@@ -1,5 +1,48 @@
-pub use massa_hash::Hash;
-pub use massa_signature::KeyPair;
-pub use massa_signature::PublicKey;
-pub use massa_signature::Signature;
-pub const PUBLIC_KEY_SIZE_BYTES: usize = massa_signature::PUBLIC_KEY_SIZE_BYTES;
+use std::fmt::Debug;
+use std::fmt::Display;
+use std::hash::Hash;
+
+use crate::error::PeerNetResult;
+
+// pub const PUBLIC_KEY_SIZE_BYTES: usize = massa_signature::PUBLIC_KEY_SIZE_BYTES;
+
+pub trait PeerNetKeyPair<K>: Send + Sync + Clone + Debug + Display + 'static
+where
+    K: PeerNetPubKey,
+{
+    fn get_public_key(&self) -> K;
+    fn sign(&self, hash: &impl PeerNetHasher) -> PeerNetResult<Vec<u8>>;
+
+    //fn to_bytes(&self) -> Vec<u8>;
+    // fn from_bytes(bytes: &[u8]) -> Self;
+}
+
+/// Trait to implement with generic ID
+pub trait PeerNetId: PartialEq + Eq + Hash + Debug + Clone + Send + Sync + 'static {
```

Comment:
While I'm developing the MassaSide I find it more useful to have a template type there constraint with PeerNetPubKey and so you can use this type as parameter of from_public_key that will avoid us to make a byte conversion

Reply:
You have it here : https://github.com/massalabs/PeerNet/blob/1398910da1d6b6bff20a147013af86d81bc9d18a/src/types.rs#L31

Reply:
We spoke about it on discord would be nice to do the same for Hashes also